src/mem_map.rs

```
@@ -115,7 +130,29 @@ fn qmp_get_mtree_stream<S: Read + Write + Clone>(stream: S) -> Result<Vec<Mappin
         })
         .map_err(|err| Error(ErrorOrigin::Connector, ErrorKind::Configuration).log_error(err))?;
 
-    Ok(qmp_parse_mtree(&mtreestr))
+    let mut mappings = qmp_parse_mtree(&mtreestr);
+    for mapping in &mut mappings {
+        if let Ok(line) = qmp.execute(&qmp::human_monitor_command {
+            command_line: format!("gpa2hva {:#10x}", mapping.range_start),
+            cpu_index: None,
+        }) {
+            let line = line.trim_end();
+            let hva = scan_fmt!(line, "Host virtual address for 0x{x} (pc.ram) is 0x{x}", [hex umem], [hex umem])
+                .map_err(|err| Error(
+                    ErrorOrigin::Connector,
+                    ErrorKind::UnsupportedOptionalFeature,
+                ).log_info(format_args!("failed to parse host address from {line:?}: {err}")));
+            match hva {
+                Ok((gpa, hva)) if gpa == mapping.range_start =>
+                    mapping.host_start = Some(hva),
+                _ => {
+                    log::warn!("failed to parse host address from {line:?}");
```

Does this ever happen? If it does happen rarely, it might be better to throw an error, instead of silently ignoring and falling back to the largest map offset.